corehq/apps/app_manager/fixtures/mobile_ucr.py

```diff
@@ -18,18 +17,29 @@
 from corehq.apps.app_manager.dbaccessors import get_apps_in_domain
 
 
-class ReportFixturesProvider(object):
+def _should_sync(restore_state):
+    last_sync_log = restore_state.last_sync_log
+    if not last_sync_log or restore_state.overwrite_cache:
+        return True
+
+    sync_interval = restore_state.restore_user.get_mobile_ucr_sync_interval()
+    return not last_sync_log or (
+        sync_interval and (datetime.utcnow() - last_sync_log.date).total_seconds() > sync_interval
+    )
+
+
+class ReportFixturesProvider(FixtureProvider):
     id = 'commcare:reports'
 
-    def __call__(self, restore_user, version, last_sync=None, app=None):
+    def __call__(self, restore_state):
         """
         Generates a report fixture for mobile that can be used by a report module
         """
-        assert isinstance(restore_user, OTARestoreUser)
-
-        if not toggles.MOBILE_UCR.enabled(restore_user.domain):
+        restore_user = restore_state.restore_user
+        if not toggles.MOBILE_UCR.enabled(restore_user.domain) or not _should_sync(restore_state):
             return []
```

Comment:
Will the mobile keep around the previously synced report fixtures if the latest sync doesn't include any?